src/js/actions/index.js

```diff
@@ -426,16 +426,21 @@ export function fetchPurchaseOrderStatuses() {
   };
 }
 
-export function fetchPaymentTerms() {
-  return (dispatch) => {
-    purchaseOrderApi.getPaymentTerms().then((res) => {
-      dispatch({
-        type: FETCH_PAYMENT_TERMS,
-        payload: res.data.data,
-      });
-    });
+export const fetchPaymentTerms = ({ translate }) => async (dispatch) => {
+  const blankPaymentTermOption = {
+    id: 'null',
+    value: 'null',
+    label: translate?.(
```

Comment:
unfortunately I'm afraid that it will most of the times fallback to the default value, due to differences in `translate` being already properly set while running into this function + due to translations not being fetched yet (I've run into the same issue with the shipment types when trying to do it like this).
I'd consider adding this to the backend where you return the JSON.
Plus I'm wondering why you have both `id` and `value` as `'null'` (as string).

Reply:
you are right about the fallback to default, the only solution for that would be to add this as an option on the API.
Thanks for catching that.

`'nulll'` is something I chose as an option value to filter Purchase orders that don't have a paymentTerm _( that is where `paymentTerm = null`)_, do you have any better suggestions for that ?

Reply:
how about doing just `null`? Does it throw any null pointers then? If so, maybe you tell where and we could figure out how to handle that together?
Basically "I'm into" doing it as `null`, but was just wondering why it has to be as string, not directly `null`.

Reply:
having it being a string instead of a null value in my opinion is safer because having it as a `null` value can result in unexpected behavior since null value is sometimes treated as an empty string or may not be included at all.

For the current example, when passing a parameter `paymentTerm: null` it sends a following request to the server
```
openboxes/api/purchaseOrdersdestination=destID&max=10&offset=0&order=desc&sort=dateOrdered&paymentTerm
```
which is being received on the backend as
```groovy
params = [
    paymentTerm: "",
    max: 10,
    offset: 0,
    ...
]
```
and later when performing a search in the `paymenTerm` param list on 
https://github.com/openboxes/openboxes/blob/3399abec63f43369d63795b3d9471522004c626c/grails-app/services/org/pih/warehouse/order/OrderService.groovy#L111-L113

the `if(...)` result in `false` even if the option is found because empty string is **falsey**
I could just do 
```groovy
if (params.list("paymentTerm").find{ it.equalsIgnoreCase("")} != null) { 
     isNull("paymentTerm") 
 } 
```
but I feel it just complicates the readability and behaviour of the code

Reply:
thanks for that explanation, this is what I expected! So, I'm fine with that, as I don't see a more reasonable and cleaner solution for that.